### PR TITLE
Refactor the way cards are laid out -- separated by expansion.

### PR DESCRIPTION
**Pull request by kberg**
Whoo that took a ton of work but I cleaned up my repo a bit.

### CODE REVIEW

**Comment by kberg** [comment]
Sorry, didn't realize this had conflicts. Pushed an update that should make this mergeable.

File: src/Game.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ import {IColony} from "./colonies/Colony";
 import {ColonyDealer, getColonyByName} from "./colonies/ColonyDealer";
 import {PlayerInterrupt} from "./interrupts/PlayerInterrupt";
 import {SelectOcean} from "./interrupts/SelectOcean";
-import {SelectResourceCard} from "./interrupts/SelectResourceCard";
+import { SelectResourceCard } from "./inputs/SelectResourceCard";
```
Comment (vincentneko) [inline src/Game.ts:38]: `SelectResourceCard `is not a playerInput, it's a `PlayerInterrupt` and should remain in interrupts folder.

File: src/Game.ts [hunk 2]
```diff
@@ -543,15 +514,6 @@ export class Game implements ILoadable<SerializedGame, Game> {
...
 
-    public addSelectResourceCardInterrupt(player: Player, count: number = 1, resourceType: ResourceType, resourceCards: Array<ICard>, title?: string): void {
-      if (resourceCards.length === 1) {
-        player.addResourceTo(resourceCards[0], count);
-        LogHelper.logAddResource(this, player, resourceCards[0], count);
-        return undefined;            
-      }
-      this.addInterrupt(new SelectResourceCard(player, this, resourceType, resourceCards, title));
-    }
-
```
Comment (vincentneko) [inline src/Game.ts:554]: Theses changes should not be mixed with the primary goal of this PR, can you please revert to initial state ?

File: src/Game.ts [hunk 3]
```diff
@@ -564,14 +526,7 @@ export class Game implements ILoadable<SerializedGame, Game> {
...
       }
-
-      if (resourceCards.length === 1) {
-        player.addResourceTo(resourceCards[0], count);
-        LogHelper.logAddResource(this, player, resourceCards[0], count);
-        return undefined;            
-      }
-
-      this.addInterrupt(new SelectResourceCard(player, this, resourceType, resourceCards, title, count));
+      this.addInterrupt({player: player, playerInput: SelectResourceCard.newInstance(this, player, resourceType, resourceCards, count, title)});
```
Comment (vincentneko) [inline src/Game.ts:529]: Theses changes should not be mixed with the primary goal of this PR, can you please revert to initial state ?

File: src/inputs/SelectResourceCard.ts [hunk 4]
```diff
@@ -1,34 +1,23 @@
...
-    public playerInput: PlayerInput;
-    constructor(
-        public player: Player,
-        public game: Game,
-        public resourceType: ResourceType,
-        public resourceCards: Array<ICard>,
-        public title: string | undefined,
-        public count: number = 1
-    ){
+export class SelectResourceCard {
```
Comment (vincentneko) [inline src/inputs/SelectResourceCard.ts:7]: Theses changes should not be mixed with the primary goal of this PR, can you please revert to initial state ?

**Review by vincentneko — CHANGES_REQUESTED** [review]
This PR looks great, the changes are huge and would need testing.
There are some modifications for `selectResourceInterrupt` that I would like to address separately, can you please revert everything related to it so we can focus on the card refactor ?

File: src/inputs/SelectResourceCard.ts (hunk 4 above)
Comment (kberg) [reply src/inputs/SelectResourceCard.ts:7]: Man. What a rookie. I am. :D

File: src/Game.ts (hunk 1 above)
Comment (kberg) [reply src/Game.ts:38]: Of course. Mea culpa. I'm sorry.

**Review by kberg — COMMENTED** [review]
I'm going to dump this and start again. Something's gone a bit pear-shaped with this branch.

**Pull request closed without merging**